web/src/routes/SearchPage.tsx

```diff
@@ -64,31 +76,38 @@ const SearchPage = ({ city, cityCode, languageCode, pathname }: CityRouteProps):
     languageCode,
   }
 
+  const handleFilterTextChanged = (filterText: string): void => {
+    setFilterText(filterText)
+    const appendToUrl = filterText.length !== 0 ? `?query=${filterText}` : ''
+    navigate(`${pathname}/${appendToUrl}`, { replace: true })
+  }
+
+  const pageTitle = `${t('pageTitle')} - ${city.name}`
+
   if (loading) {
     return (
       <CityContentLayout isLoading {...locationLayoutParams}>
+        <Helmet pageTitle={pageTitle} languageChangePaths={languageChangePaths} cityModel={city} />
+        <SearchInput
+          filterText={filterText}
+          placeholderText={t('searchPlaceholder')}
+          onFilterTextChange={handleFilterTextChanged}
+          spaceSearch
+        />
         <LoadingSpinner />
       </CityContentLayout>
     )
   }
 
-  if (!searchResults) {
-    const error = categoriesError || new Error('Categories should not be null!')
+  if (error) {
+    reportError(error)
```

Comment:
Errors should not be reported here, otherwise we report an error on every render. In general we should have this already covered.
```suggestion
```